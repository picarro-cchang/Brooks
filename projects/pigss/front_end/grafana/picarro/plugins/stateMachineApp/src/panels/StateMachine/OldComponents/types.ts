// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
