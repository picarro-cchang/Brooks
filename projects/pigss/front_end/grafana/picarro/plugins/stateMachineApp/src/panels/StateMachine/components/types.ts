export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
