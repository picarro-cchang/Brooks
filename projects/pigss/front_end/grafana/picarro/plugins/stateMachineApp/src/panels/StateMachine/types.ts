import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
