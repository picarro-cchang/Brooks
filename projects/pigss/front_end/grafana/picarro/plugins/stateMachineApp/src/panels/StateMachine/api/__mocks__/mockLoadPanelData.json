{
    "max_steps": 32,
    "panel_to_show": 2,
    "current_step": 1,
    "focus": {
        "row": 1,
        "column": 2
    },
    "last_step": 1,
    "steps": {
        "1": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 1
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
        }
    },
    "num_plan_files": 1,
    "plan_filename": "test.pln",
    "plan_files": {
        "1": "test.pln"
    },
    "bank_names": {
        "1": {
            "name": "B1",
            "channels": {
                "1": "Channel 1",
                "2": "Channel 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
        },
        "2": {
            "name": "B1",
            "channels": {
                "1": "Channel 1",
                "2": "Channel 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
        },
        "3": {
            "name": "Bank 2",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
        },
        "4": {
            "name": "Bank 3",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
        }
    }
}