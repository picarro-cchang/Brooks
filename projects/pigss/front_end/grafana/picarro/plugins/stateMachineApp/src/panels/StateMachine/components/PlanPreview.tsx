import React, { PureComponent, ReactText } from "react";
import ReactList from "react-list";
import { PlanLoadPanelOptions, Plan } from "./../types";

interface State {
  plan: Plan;
}
class PlanPreviewPanel extends PureComponent<PlanLoadPanelOptions, State> {
  constructor(props) {
    super(props) 
    this.state = {
      plan: this.props.plan,
    }
  }


  render() {
    return (
      <div className="panel-save">
        Hello! Showing plan {this.props.plan.plan_filename}
        <button
            onClick={e => this.props.updatePanel(0)}
        >
            Ok
        </button>
        <button
            onClick={e => this.props.updatePanel(0)}
        >
            Cancel
        </button>
      </div>
    );
  }
}

export default PlanPreviewPanel;
