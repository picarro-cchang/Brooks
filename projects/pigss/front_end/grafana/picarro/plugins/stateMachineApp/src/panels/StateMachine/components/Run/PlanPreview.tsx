import React, { PureComponent, ReactText } from "react";
import ReactList from "react-list";
import {
  LoadPanelCommandOptions,
  Plan,
  PlanStep,
  PreviewPanelOptions,
} from "../types";

interface State {
  plan: Plan;
}
class PlanPreview extends PureComponent<PreviewPanelOptions, State> {
  constructor(props) {
    super(props);
    this.state = {
      plan: this.props.plan,
    };
  }

  makePlanRow = (row: number) => {
    let portString = "";
    let durationString = "";
    const planRow = this.props.plan.steps[row] as PlanStep;
    if (planRow.reference != 0) {
      portString = "Reference";
    } else {
      for (const bank in planRow.banks) {
        if (planRow.banks.hasOwnProperty(bank)) {
          const bank_name = this.props.plan.bank_names[bank].name;
          const bank_config = planRow.banks[bank];
          if (bank_config.clean != 0) {
            portString = `Clean ${bank_name}`;
            break;
          } else if (bank_config.chan_mask != 0) {
            const mask = bank_config.chan_mask;
            // Find index of first set bit using bit-twiddling hack
            const channel = (mask & -mask).toString(2).length;
            const ch_name = this.props.plan.bank_names[bank].channels[channel];
            const bankNum = Number(bank);
            const portNumber = (bankNum - 1) * 8 + channel;
            portString = portNumber + ": " + ch_name;
            break;
          }
        }
      }
    }
    if (planRow.duration !== 0) {
      durationString = `${planRow.duration}`;
    }

    return (
      <div className="gf-form-inline" key={row}>
        <div
          className="panel-plan-text"
          style={{ fontSize: "17px", marginTop: "5px" }}
        >
          {row < 10 ? (
            <>{row + ". "}</>
          ) : (
            <span
              className="panel-plan-text"
              style={{
                marginLeft: "-7px",
                marginRight: "7px",
                marginTop: "5px",
                fontSize: "17px",
              }}
            >
              {row + ". "}
            </span>
          )}
          {portString} Duration: {durationString}
        </div>
      </div>
    );
  };

  renderItem = (index: number, key: ReactText) => (
    <div key={key}>{this.makePlanRow(index + 1)}</div>
  );

  render() {
    return (
      <div className="panel-plan-preview">
        <h4>Plan: {this.props.plan.plan_filename}</h4>
        <h6>
          Press Ok to continue with this plan, or Cancel to choose another.
        </h6>
        <div className="panel-preview-inner">
          <div style={{ marginLeft: "5px" }}>
            <ReactList
              itemRenderer={this.renderItem}
              length={this.props.plan.last_step}
              type={"uniform"}
            />
          </div>
        </div>
        <div className="row btn-row-2">
          <div>
            <button
              id={"cancel-load-plan"}
              className={"btn btn-block btn-cancel btn-group-preview"}
              onClick={(e) => {
                this.props.ws_sender({
                  element: "filename_cancel",
                });
                this.props.cancelLoadPlan();
                this.props.updatePanel(1);
              }}
            >
              Cancel
            </button>
          </div>
          <div>
            <button
              className={"btn btn-block btn-green btn-group-preview"}
              onClick={(e) => {
                this.props.ws_sender({
                  element: "filename_ok",
                });
                this.props.setModalInfo(
                  true,
                  `<div>Load File ${this.state.plan.plan_filename} for running?</div>`,
                  2,
                  {
                    1: {
                      caption: "Ok",
                      className: "btn btn-success btn-large",
                      response: {
                        plan: this.state.plan,
                        name: this.state.plan.plan_filename,
                      },
                    },
                    2: {
                      caption: "Cancel",
                      className: "btn btn-danger btn-large",
                      response: null,
                    },
                  },
                  "loadPlan"
                );
              }}
            >
              Ok
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default PlanPreview;
