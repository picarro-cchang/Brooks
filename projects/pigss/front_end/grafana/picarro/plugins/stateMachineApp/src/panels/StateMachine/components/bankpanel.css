.bank-btn {
    border-radius: 25%;
    /*border: 4px solid #bbbbbbbd;*/
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    height: 90px;
    width: 90px;
    color: black;
    position: relative;
}
.chn-label {
    position: absolute;
    top: 20px;
    left: -0.37em;
    bottom: 0;
    right: 0;
}
.btn-clean {
    width:100%;
    border: 3px solid hsla(0, 0%, 100%, 0.34);
    color: black;
    border-radius: 10px;
}
.btn-clean-act {
    border: none;
}

.test {
    color: black;
}
.btn-group {
    width: 100%;
    /*height: 35px;*/
    border-radius: 10%;
    text-align: center;
    /*margin-right: 20px;*/
    margin-left: -20px;
    font-size: 17px;
}

.btn-row-1 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-2 {
    margin-top: 10px;
    align-items: center;
    display: flex;
    justify-content: center;
}
.btn-row-3 {
    margin-top: 10px;
    align-items: center;
    style: flex;
    justify-content: center;
}
.panel-plan {
    height: 640px;
    /*background-color: #888;*/
    /*border: 3px solid #111;*/
    /*border-radius: 7px;*/
    /*background-color: #82828287;*/
    background-color: #cacaca;
    padding: 10px;
    /*border: 2px solid #1111117a;*/
    border-radius: 5px;
}
.panel-plan-inner {
    overflow-x: hidden;
    overflow-y: auto;
    max-height: 420px;
    border-radius: 3px;
}
.plan-input {
    background-color: white;
    border: 1px solid black;
    border-radius: 2px;
    /*margin-right: -20px;*/
    color: black;
    margin-bottom: 5px;
}
.panel-bank {
    /*border: 2px solid #111;*/
    border: 2px solid #1111117a;
    margin-right: 2%;
    border-radius: 10px;
    padding:10px;
    float: left;
    background-color: rgba(136, 136, 136, 0.51);

}
.grid-bank {
    display: grid;
    grid-gap: 10px;
    grid-template-columns: 1fr 1fr;
    padding-bottom: 10px;
}
.btn-green {
    background-color:  rgb(86, 166, 75);
}
.btn-1-3 {
    grid-column-start: 1;
    grid-column-end: 4;
    width: 100%;
}
.reference-btn {
    display: inline-block;
}
.ref-btn {
    width: 400px;
    grid-row-start: 2;
    grid-column-start: 1;
    /*margin-top: -100px;*/
}

.btn-command {
    border-radius: 5px;
    color: black;
    height: 80px;
}
.btn-edit {
    border-radius: 5px;
    color: white;
    height: 55px;
    width: 100px;
    margin-right: 20px;
    background-color: red;
    background-image: none;
}
.btn-cancel {
    background-image: none;
    background-color: red;
}
.edit-panel {
    color: black;
    background: white;
    border: 1px solid black;
    border-radius: 2px;
}
.edit-label {
    margin-left: 30px;
    margin-right: 10px;
    font-size: 15px;
    color: black;
}
.panel-edit {
    height: 550px;
    background-color: #cacaca;
    padding: 10px;
    border: 3px solid #111;
    border-radius: 7px;
}
.btn-edit-panel {
    margin-bottom: 10px;
    width: 95px;
    height: 40px;
    border-radius: 5px;
    text-align: center;
    font-size: 20px;
}
.btn-run {
    width: 100%;
}
.btn-inverse.disabled {
    background-color: #333;
    opacity: .9;
}
.btn-success {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;
}

.btn-success:active, .btn-success:focus {
    /*background-image: -webkit-linear-gradient(top,#6fe28c,#23843b);*/
    background-color: green;

}
.btn-run-active {
    background-color: rgba(98, 100, 104, 0.86);
}
.btn-run-issue {
    background-color: #716e6e;
}
.btn-chann {
     background-color:  rgb(86, 166, 75);
 }
.col-bank-name {
    /*max-width: 45%;*/
    /*min-width: 20%;*/
    text-align:left;
    padding-left: 5px;
    padding-right: 5px;
}
.input-small {
    width: 70px;
    margin-left: -10px;
}

.radio-btn {
    display: block;
    position: relative;
    /*padding-left: 35px;*/
    margin-bottom: 12px;
    /*margin-left: 20px;*/
    cursor: pointer;
    font-size: 22px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
}
.radio-btn input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
}

.checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 25px;
    width: 25px;
    background-color: #eee;
    border-radius: 50%;
}

.radio-btn:hover input ~ .checkmark {
    background-color: #565657;
}

.radio-btn input:checked ~ .checkmark {
    background-color: rgb(31, 96, 196);
}


.checkmark:after {
    content: "";
    position: absolute;
    display: none;
}

.radio-btn input:checked ~ .checkmark:after {
    display: block;
}

.radio-btn .checkmark:after {
    top: 9px;
    left: 9px;
    width: 8px;
    height: 8px;
    border-radius: 50%;
    background: white;
}
.button-edit {
    justify-content: center;
}

.cancel:before {
    content: '\2716';
}

.cancel {
    position: absolute;
    top: 0;
    right: 10%;
    cursor: pointer;
    color: black;
    font-size: 25px;
    display: inline-block;
}

