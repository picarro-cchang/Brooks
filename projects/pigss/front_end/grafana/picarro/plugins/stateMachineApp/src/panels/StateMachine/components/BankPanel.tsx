import React, { PureComponent } from "react";
import "./bankpanel.css";

export interface BankPanelOptions {
  bank: number;
  uistatus: {
    bank?: { [bankNum: string]: string };
    clean?: { [bankNum: string]: string };
    channel?: { [bankNum: string]: { [channelNum: string]: string } };
  };
  ws_sender: (o: object) => void;
  plan: {
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
  };
}

class BankPanel extends PureComponent<BankPanelOptions> {
  bankStyleOpt = {
    READY: { color: "#424242", backgroundColor: "#e4e4e4" },
    ACTIVE: { color: "#fff", backgroundColor: "#56a64b" },
    CLEAN: { color: "#fff", backgroundColor: "#4BBEE3" },
    REFERENCE: { color: "#440", backgroundColor: "#e0b400" }
  };
  cleanClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    ACTIVE: "btn-light",
    CLEAN: "btn-secondary btn-clean-act"
  };
  channelClassNameOpt = {
    DISABLED: "btn-inverse-2 disabled",
    READY: "btn-ready",
    AVAILABLE: "btn-ready",
    ACTIVE: "btn-green",
    CLEAN: "btn-light",
    REFERENCE: "btn-warning"
  };

  render() {
    let bankStyle = {};
    let cleanClassNames = "";
    let cleanDisabled = true;
    let getChannelClassNames = (_: number) => "";
    let getChannelDisabled = (_: number) => true;
    let test = {};

    if ("bank" in (this.props.uistatus as any)) {
      const bankStatus: string = this.props.uistatus.bank && (this.props.uistatus.bank as any)[
        this.props.bank
      ];
      const channelStatus: { [key: number]: string } = this.props.uistatus.channel && (this.props.uistatus
        .channel as any)[this.props.bank];
      const cleanStatus: string = this.props.uistatus.clean && (this.props.uistatus.clean as any)[
        this.props.bank
      ];
      bankStyle = (this.bankStyleOpt as any)[bankStatus];
      cleanClassNames = (this.cleanClassNameOpt as any)[cleanStatus];
      cleanDisabled = cleanStatus !== "READY";
      getChannelClassNames = chan => {
        return this.channelClassNameOpt !== undefined && 
                channelStatus !== undefined && 
                (this.channelClassNameOpt as any)[(channelStatus as any)[chan]];
      }
      getChannelDisabled = chan => { return channelStatus !== undefined && (channelStatus as any)[chan] !== "READY" };
      test = channelStatus;
    }

    const channelButtons = [];
    for (let i = 1; i <= 8; i++) {
      channelButtons.push(
        getChannelDisabled(i) ? (
          <button
            key={i}
            id={"channel-" + i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
            style={{ color: "black" }}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}
              </u>
            </p>
            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}{" "}
            </p>
          </button>
        ) : (
          <button
            onClick={e => {
              this.props.ws_sender({
                element: "channel",
                bank: this.props.bank,
                channel: i
              });
            }}
            id={"channel-" + i}
            disabled={getChannelDisabled(i)}
            key={i}
            className={"btn btn-large bank-btn " + getChannelClassNames(i)}
          >
            <p className="chn-label">
              <u className={"chn-name-" + i}>
                {this.props.plan.bank_names[this.props.bank].channels[i]}{" "}
              </u>
            </p>

            <p id={"chn-status-" + i} className={"chn-status"}>
              {" "}
              {test && test[i]}
            </p>
          </button>
        )
      );
    }

    const cleanButton = cleanDisabled ? (
      <div className={"btn btn-large btn-clean " + cleanClassNames}>
        {"Clean"}
      </div>
    ) : (
      <button
        id="clean"
        onClick={e =>
          this.props.ws_sender({ element: "clean", bank: this.props.bank })
        }
        className={"btn btn-large btn-clean " + cleanClassNames}
      >
        {"Clean"}
      </button>
    );
    const value: string = this.props.plan.bank_names[this.props.bank].name;
    return (
      <div>
        <div className="panel-bank" style={bankStyle}>
          <div style={{ width: "100%" }}>
            <h2 style={{ color: "#424242", marginBottom: "10px" }}>{value}</h2>
            <div className="grid-bank">{channelButtons}</div>
            {cleanButton}
          </div>
        </div>
      </div>
    );
  }
}
export default BankPanel;
