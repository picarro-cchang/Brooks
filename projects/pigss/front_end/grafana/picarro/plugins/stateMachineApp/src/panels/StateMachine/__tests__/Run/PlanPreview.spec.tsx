import React, { ReactText } from "react";
import { shallow, mount, render } from "enzyme";
import WS from "jest-websocket-mock";
import "jest-styled-components";
import PlanPreview from "./../../components/Run/PlanPreview";
import { LoadPanelCommandOptions, PreviewPanelOptions } from "./../../components/types";
import mockLoadPanelData from "./../../api/__mocks__/mockLoadPanelData.json";

const mockClick = jest.fn(element => {
  return element;
});
const mockLoadPlan = jest.fn()
const mockDeleteFile = jest.fn();
const mockUpdatePanel = jest.fn();
const mockCancelLoadPlan = jest.fn();
const mockSetModalInfo = jest.fn();

const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;
const defaultProps: PreviewPanelOptions = {
  plan: mockLoadPanelData,
  updatePanel: mockUpdatePanel,
  fileNames: ["Test1","Test2"],
  cancelLoadPlan: mockCancelLoadPlan,
  ws_sender: mockClick,
  loadedFileName: "Testing",
  setModalInfo: mockSetModalInfo
};

describe("<PlanPreview />", () => {
  const wrapper = shallow(<PlanPreview {...defaultProps} />);
  const instance = wrapper.instance() as PlanPreview;

  it("Match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders Items", () => {
    instance.renderItem = jest.fn();
    instance.renderItem(0, 0);
    expect(instance.renderItem).toHaveBeenCalled();

    const fileList = mount(<PlanPreview {...defaultProps} />).find(
      "ReactList"
    );
    expect(fileList).toMatchSnapshot();
    expect(
      fileList
        .find("div")
        .at(1)
        .text()
    ).toEqual("1. 1: Channel 1 Duration: 30");
  });

  it("Cancel", () => {

  });

  it("Ok", () => {

  });

});
