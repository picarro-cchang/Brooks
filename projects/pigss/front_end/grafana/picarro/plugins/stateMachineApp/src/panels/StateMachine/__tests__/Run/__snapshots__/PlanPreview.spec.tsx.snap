// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanPreview /> Match Snapshot 1`] = `
<div
  className="panel-plan-preview"
>
  <h4>
    Plan: 
    test.pln
  </h4>
  <h6>
    Press Ok to continue with this plan, or Cancel to choose another.
  </h6>
  <div
    className="panel-preview-inner"
  >
    <div
      style={
        Object {
          "marginLeft": "5px",
        }
      }
    >
      <ReactList
        axis="y"
        itemRenderer={[Function]}
        itemsRenderer={[Function]}
        length={3}
        minSize={1}
        pageSize={10}
        scrollParentGetter={[Function]}
        scrollParentViewportSizeGetter={[Function]}
        threshold={100}
        type="uniform"
        useStaticSize={false}
        useTranslate3d={false}
      />
    </div>
  </div>
  <div
    className="row btn-row-2"
  >
    <div>
      <button
        className="btn btn-block btn-cancel btn-group-preview"
        id="cancel-load-plan"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
    <div>
      <button
        className="btn btn-block btn-green btn-group-preview"
        onClick={[Function]}
      >
        Ok
      </button>
    </div>
  </div>
</div>
`;

exports[`<PlanPreview /> Renders Items 1`] = `
<ReactList
  axis="y"
  itemRenderer={[Function]}
  itemsRenderer={[Function]}
  length={3}
  minSize={1}
  pageSize={10}
  scrollParentGetter={[Function]}
  scrollParentViewportSizeGetter={[Function]}
  threshold={100}
  type="uniform"
  useStaticSize={false}
  useTranslate3d={false}
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTransform": "translate(0px, 0px)",
          "msTransform": "translate(0px, 0px)",
          "transform": "translate(0px, 0px)",
        }
      }
    >
      <div>
        <div
          key="0"
        >
          <div
            className="gf-form-inline"
            key="1"
          >
            <div
              className="panel-plan-text"
              style={
                Object {
                  "fontSize": "17px",
                  "marginTop": "5px",
                }
              }
            >
              1. 
              1: Channel 1
               Duration: 
              30
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</ReactList>
`;
